test/jdk/java/awt/print/PrinterJob/raster/RasterTest.java

```diff
@@ -21,248 +21,146 @@
  * questions.
  */
 
-/**
+import java.awt.Button;
+import java.awt.Canvas;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Frame;
+import java.awt.GradientPaint;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.image.BufferedImage;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+
+import jtreg.SkippedException;
+
+/*
  * @test
  * @bug 4242639
  * @summary Printing quality problem on Canon and NEC
  * @key printer
+ * @library /java/awt/regtesthelpers
+ * @library /test/lib
+ * @build PassFailJFrame
+ * @build jtreg.SkippedException
  * @run main/manual RasterTest
  */
-import java.awt.*;
-import java.awt.geom.*;
-import java.awt.event.*;
-import java.awt.print.*;
-import java.awt.Toolkit;
-import java.awt.image.BufferedImage;
-
-
 public class RasterTest extends Frame implements ActionListener {
 
- private RasterCanvas c;
-
- public static void main(String args[]) {
-  String[] instructions =
-        {
-         "You must have a printer available to perform this test",
-         "This test uses rendering operations which force the implementation",
-         "to print the page as a raster",
-         "You should see two square images, the 1st containing overlapping",
-         "composited squares, the lower image shows a gradient paint.",
-         "The printed output should match the on-screen display, although",
-         "only colour printers will be able to accurately reproduce the",
-         "subtle color changes."
-       };
-      Sysout.createDialog( );
-      Sysout.printInstructions( instructions );
-
-    RasterTest f = new RasterTest();
-    f.show();
- }
+    private final RasterCanvas c;
+
+    private static final String instructions =
+            "You must have a printer available to perform this test\n" +
+            "This test uses rendering operations which force the implementation\n" +
+            "to print the page as a raster\n" +
+            "You should see two square images, the 1st containing overlapping\n" +
+            "composited squares, the lower image shows a gradient paint.\n" +
+            "The printed output should match the on-screen display, although\n" +
+            "only colour printers will be able to accurately reproduce the\n" +
+            "subtle color changes.";
```

Comment:
When saved to pdf the gradient square color looks different from the one in the test window. Probably it works as expected on a color printer.